### ISSUE TITLE
Defining less colours than segments on a CleanSpline with .BlendExt throws error

### ISSUE THREAD

**Issue by bakumoe** [JujuAdams/Clean-Shapes#22]
I mentioned this in the pull request, the spline test room was crashing due to it.

Assuming x1, y1, etc. are set, the following code would compile:
```
CleanSpline([x1, y1, x2, y2, x3, y3, x4, y4, x5, y5], 6) // more segments than colours
.BlendExt([c_white, 1.0, c_yellow, 1.0, c_fuchsia, 1.0, c_aqua, 1.0, c_lime, 1.0])
.Draw();
```

But this would throw an ungraceful error:
```
CleanSpline([x1, y1, x2, y2, x3, y3, x4, y4, x5, y5], 5) // same amount or less segments than colours
.BlendExt([c_white, 1.0, c_yellow, 1.0, c_fuchsia, 1.0, c_aqua, 1.0, c_lime, 1.0])
.Draw();
```

```
___________________________________________
############################################################################################
ERROR in
action number 1
of Draw Event
for object oTestSpline:

Push :: Execution Error - Variable Index [10] out of range [10] - -15.argument0(91,10)
 at gml_Script_anon___CleanClassSpline_gml_GlobalScript_CleanSpline_5301___CleanClassSpline_gml_GlobalScript_CleanSpline (line 201) -             var _colour = merge_colour(_inputArray[_int], _inputArray[_int + 2], _frac);
############################################################################################
gml_Script_anon___CleanClassSpline_gml_GlobalScript_CleanSpline_5301___CleanClassSpline_gml_GlobalScript_CleanSpline (line 201)
gml_Object_oTestSpline_Draw_0 (line 1) - CleanSpline([x1, y1, x2, y2, x3, y3, x4, y4, x5, y5], 5)

```

**Comment by JujuAdams** [comment]
This should error, yes. I'll put in a more helpful error message.

**Comment by JujuAdams** [comment]
Oh wait, a _spline_! Yeah, that's a bug then

**Comment by JujuAdams** [comment]
Fixed for v1.0.10
